### ISSUE TITLE
Multiple values of categorical feature associated with single time point

### ISSUE THREAD

**Issue by PeggyFan** [awslabs/gluon-ts#548]
Is it possible to input data where you have multiple values (a categorical variable) associated with a timestamp? Say I want to make forecast on a group of product (rather than individual product) and the group contains multiple values for a product feature I wish to use in the model.

For example:
`[{date: '2020-01-01', id: '001', product_feature: ['red', 'blue'], sale: 300},
{date: '2020-01-01', id: '002', product_feature: ['yellow', 'blue', 'green'], sale: 500},
{date: '2020-01-01', id: '003', product_feature: ['orange'], sale: 100},
{date: '2020-01-01', id: '004', product_feature: ['orange', 'green'], sale: 90}]`

Thank you!

**Comment by mbohlkeschneider** [comment]
Hi @PeggyFan, 

Are you asking for a categorical features that change over time? Then you are referring to `feat_dynamic_cat` features. These are currently not supported by GluonTS models, as discussed [here](#392). 

If the categorical features you are thinking about are static features, you can use the `feat_static_cat` field. Please refer to the [extended forecasting tutorial](https://github.com/awslabs/gluon-ts/blob/master/docs/examples/extended_forecasting_tutorial/extended_tutorial.md) on how to use them.

**Comment by PeggyFan** [comment]
@mbohlkeschneider Thank you. They are static features. My next question (since it is not obvious from the extended forecasting tutorial) is if I can put multiple rows of the same timestamp to account for the multiple features. Would this throw an error/affect the prediction?

For example, 
`[
{date: '2020-01-01', id: '002', product_feature: 'yellow', sale: 300},
{date: '2020-01-01', id: '002', product_feature: 'blue', sale: 300},
{date: '2020-01-01', id: '002', product_feature: 'green', sale: 300}
]`

**Comment by mbohlkeschneider** [comment]
Hi @PeggyFan,

Your snippet does not show valid GluonTS formatting. 

You can encode multiple features like this:

`[{"start" : "2020-01-01', "target": [100, 200, 300, 100], "feat_static_cat" : [1, 2, 10]}]`

This represents a single time series with `start` representing the start date, `target` being the time series values, and `feat_static_cat` are the label encodings of (multiple) features, there the i-th entry in the list represents the label encoding of a single feature. 

However, there is no automatic what to label encode the features or aggregate the data. This has to be done by the user before passing data to GluonTS. 

Hope that was clear!

**Comment by PeggyFan** [comment]
@mbohlkeschneider Thank you. I think I understand it now. One last question- for `feat_static_cat` the encoding has to be numeric? So for say a list of features colors : ['red', 'yellow', 'green'], it has to be something like [1,2,3]? Does the numbers creates a relationship that doesn't exist among the categorical feature values? for example, 3 > 1 but there's nothing suggesting 'green' > 'red'.
or do I need to create dummy variables for each color? (then how would the GluonTS formatting look?)

**Comment by mbohlkeschneider** [comment]
> @mbohlkeschneider Thank you. I think I understand it now. One last question- for feat_static_cat the encoding has to be numeric?

Yes. 

> So for say a list of features colors : ['red', 'yellow', 'green'], it has to be something like [1,2,3]? Does the numbers creates a relationship that doesn't exist among the categorical feature values? for example, 3 > 1 but there's nothing suggesting 'green' > 'red'.

Usually, we use label encodings and this works fine. You could use [sklearn's label encoder](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.LabelEncoder.html), for example. The label encodings get internally converted into [embeddings](https://github.com/awslabs/gluon-ts/blob/master/src/gluonts/model/deepar/_network.py#L233). 

In our experience that works quite well, though. Of course, there might be a smarter way of doing this. Let us know if you find another way of doing it!

> or do I need to create dummy variables for each color? (then how would the GluonTS formatting look?)

Yes, you have to do the label encodings yourself. sklearn's label encoder does a fine job here.

**Comment by PeggyFan** [comment]
@mbohlkeschneider thank you for your time answering my questions! this really helps!

**Comment by miteshyadav** [comment]
@PeggyFan Could you please share how did you solve this? Thanks
